### PR TITLE
Add vertically integrated tracer diagnostics

### LINKED ISSUE #182: Feature: add diagnostics for hydrometeors
It would be nice to have column integrals for all the water species separately. As far as I can tell, there are no available column diagnostics for graupel, rain, or snow. The model only supports saving ice water path "iw", liquid water path "lw",  and precipitable water "pwat". 

cc @spencerkclark

### PR DESCRIPTION
**Pull request by spencerkclark**
This PR adds vertically integrated diagnostics for the tracers used in FV3GFS.  A notebook verifying that these work can be found [here](https://github.com/VulcanClimateModeling/explore/blob/master/spencerc/2021-04-23-vertically-integrated-diagnostics/2021-04-26-vertically-integrated-tracer-diagnostics.ipynb).

Resolves #182

### CODE REVIEW

**Review by nbren12 — APPROVED** [review]
Thanks spencer. Looks great!

I'm not sure why the nix-mac tests are failing. I think it may be an intermittent issue.

**Comment by nbren12** [comment]
@rheacangeo @ofuhrer Does DSL want to review this?

File: FV3/atmos_cubed_sphere/tools/fv_diagnostics.F90 [hunk 1]
```diff
@@ -2442,6 +2452,13 @@ subroutine fv_diag(Atm, zvir, Time, print_freq)
           used = send_data(idiag%id_lw, a2*ginv, Time)
        endif
 
+       do tracer_index = 1, Atm(n)%ncnst
+         if (idiag%id_vertically_integrated_tracers(tracer_index) > 0) then
+            a2 = ginv * sum(Atm(n)%q(isc:iec,jsc:jec,1:npz,tracer_index) * Atm(n)%delp(isc:iec,jsc:jec,1:npz), dim=3)
```
Comment (ofuhrer) [inline FV3/atmos_cubed_sphere/tools/fv_diagnostics.F90:2457]: Could help for clarity to write `a2(:,:) = ...` here.

**Review by ofuhrer — APPROVED** [review]
I am no expert on the `fv_diagnostics.f90` code, but this change looks good to me.

**Comment by spencerkclark** [comment]
Thanks for having a look @ofuhrer!

> I'm not sure why the nix-mac tests are failing. I think it may be an intermittent issue.

Agreed, it is unrelated so I will go ahead and merge.

**Pull request merged**